### PR TITLE
feat: [PL-39081]: Added a param to include parent scope settings

### PR DESCRIPTION
**Pull request by tiwaritejas**
## Describe your changes
In order to solve certain existing use cases, the settings list API will now take a parameter to fetch settings from parent scopes as well. For eg, if a setting is only available at an account scope, it can be fetched at the project scope by passing a query param- "includeParentScopes" in the settings list API.

Testing:-
1. Tested that the API returns the settings allowed at the given scope as well as at its parent scopes in case the flag is passed as true.
2. Tested that only the existing scope settings are returned in case the flag is passed as false.
3. Added unit testing for the same.

## Checklist
- [x] I've documented the changes in the PR description.
- [x] I've tested this change either in PR or local environment.
- [x] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- 
  Specific builds
  - `trigger manager`
  - `trigger dms`
  - `trigger ng_manager`
  - `trigger cvng `
  - `trigger cmdlib`
  - `trigger template_svc`
  - `trigger events_fmwrk_monitor`
  - `trigger event_server`
  - `trigger change_data_capture`
  -  Trigger multiple builds together: For eg: `trigger dms, manager`
- Immutable delegate `trigger publish-delegate`
</details>

## [Latest PR Check Triggers](https://github.com/harness/harness-core/blob/develop/.github/pull_request_template.md)

<details>
  <summary>PR Check triggers</summary>
You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ut0, ut1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
  - CodeFormat: `trigger codeformat`
  - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- Trigger CommonChecks: `trigger commonchecks`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- UnitTests-ALL: `trigger uts`
- UnitTests-0: `trigger ut0`
- UnitTests-1: `trigger ut1`
- UnitTests-2: `trigger ut2`
- UnitTests-3: `trigger ut3`
- UnitTests-4: `trigger ut4`
- UnitTests-5: `trigger ut5`
- UnitTests-6: `trigger ut6`
- UnitTests-7: `trigger ut7`
- UnitTests-8: `trigger ut8`
- UnitTests-9: `trigger ut9`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- GitLeaks: `trigger gitleaks`
- Trigger all Checks: `trigger smartchecks`
- Go Build: `trigger gobuild`
- Validate_Reviews: `trigger review`
- Module Dependency Check: `trigger mdc`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/49362)

### CODE REVIEW

**Comment by tiwaritejas** [comment]
trigger uts

File: 120-ng-manager/modules/ng-settings/contracts/src/main/java/io/harness/ngsettings/SettingConstants.java [hunk 1]
```diff
@@ -29,4 +29,6 @@ public final class SettingConstants {
   public static final String GROUP_KEY = "group";
   public static final String ALLOW_EDIT = "Is the setting editable at the current scope";
   public static final String ALLOWED_SCOPES = "List of scopes where the setting is available";
+  public static final String INCLUDES_PARENT_SCOPE = "Are the child scope settings included";
```
Comment (nishantsaini20) [inline 120-ng-manager/modules/ng-settings/contracts/src/main/java/io/harness/ngsettings/SettingConstants.java:32]: The description is not clear.

**Review by nishantsaini20 — CHANGES_REQUESTED** [review]

**Review by VikasMaddukuriHarness — APPROVED** [review]

File: 120-ng-manager/modules/ng-settings/contracts/src/main/java/io/harness/ngsettings/SettingConstants.java (hunk 1 above)
Comment (tiwaritejas) [reply 120-ng-manager/modules/ng-settings/contracts/src/main/java/io/harness/ngsettings/SettingConstants.java]: Updated it to "Flag to include the settings which only exist at the parent scopes"

**Comment by tiwaritejas** [comment]
trigger uts

**Review by nishantsaini20 — APPROVED** [review]

**Pull request merged**